core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -647,6 +651,30 @@ public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings,
     }
 
     /**
+     * Verify that the contents on disk for the given index is deleted; if not, delete the contents if allowed.
+     * @param index {@code Index} to make sure its deleted from disk
+     * @param clusterState the current {@code ClusterState}
+     * @return true if the index contents is deleted on this node
+     */
+    public boolean tryDeleteIndexContents(final Index index, final ClusterState clusterState) {
+        boolean success = false;
+        if (canDeleteIndexContents(index, null, false)) {
+            try {
+                final IndexMetaData metaData = metaStateService.loadIndexState(index);
+                deleteClosedIndex("deleted index was not assigned to local node", metaData, clusterState);
+                success = true;
+            } catch (IOException e) {
+               success = false;
+            }
+        }
+        if (success == false) {
+            logger.warn("[{}] Deletion request unsuccessful, likely scenario is an index was " +
+                        "created and deleted while the node was offline.", index);
+        }
+        return success;
+    }
```

Comment:
@bleskes As we discussed earlier, I tried implementing a method that verifies and deletes index contents.  However, you can see above that I had to load the `IndexMetaData` from disk.  The reason is that the `deleteClosedIndex` method uses it.  Tracing through the code, the index metadata is mainly used to get the `Index`, but in the following method, it uses the `IndexSettings`:

`private void deleteIndexStore(String reason, Index index, IndexSettings indexSettings, boolean closed)`

This method does the `canDeleteIndexContents` check and then calls `nodeEnv#deleteIndexDirectorySafe`, which uses the `IndexSettings` to obtain locks for each shard and to see if there are any custom data dirs that need to be deleted.  

So I'm not sure we can get away with not loading the `IndexMetaData` unless we forego a lot of those checks and just do a file level delete of the index folder only without any checks or locks.